lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -255,20 +251,21 @@ def test_malformed_payload(self, client, server_config, pbench_token):
         """
         Test behavior when payload is not valid JSON
         """
-        response = self.make_request_call(
-            client,
-            server_config.rest_uri + self.pbench_endpoint,
-            {
-                "Authorization": "Bearer " + pbench_token,
-                "Content-Type": "application/json",
-            },
-            data='{"x":0]',
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert (
-            response.json.get("message")
-            == "Invalid request payload: '400 Bad Request: The browser (or proxy) sent a request that this server could not understand.'"
-        )
+        if self.api_method == API_METHOD.POST:
+            response = self.make_request_call(
+                client,
+                server_config.rest_uri + self.pbench_endpoint,
+                {
+                    "Authorization": "Bearer " + pbench_token,
+                    "Content-Type": "application/json",
+                },
+                data='{"x":0]',
+            )
+            assert response.status_code == HTTPStatus.BAD_REQUEST
+            assert (
+                response.json.get("message")
+                == "Invalid request payload: '400 Bad Request: The browser (or proxy) sent a request that this server could not understand.'"
+            )
```

Comment:
I think Dave also mentioned this, but this test now does _nothing_ if the method is not `POST`.  I concur with Dave -- this should be recast this as
```
        if self.api_method != API_METHOD.POST:
            pytest.skip("skipping " + self.test_malformed_payload.__name__)

        response = self.make_request_call(
            client,
            ...
```
Likewise for the other functions like this.